web/src/app/schedules/ScheduleShiftList.js

```diff
@@ -266,17 +267,33 @@ export default class ScheduleShiftList extends React.PureComponent {
       <TextField
         fullWidth
         label='Time Limit (days)'
-        value={Duration.fromISO(this.props.duration).as('days')}
+        value={
+          this.state.isClear
+            ? ''
+            : Duration.fromISO(this.props.duration).as('days')
+        }
         disabled={this.props.activeOnly}
         max={30}
         min={1}
         type='number'
+        onBlur={() => {
+          this.setState({ isClear: false })
+        }}
         onChange={(e) => {
-          this.props.handleSetDuration(
-            Duration.fromObject({
-              days: clamp(1, 30, parseInt(e.target.value, 10)),
-            }).toISO(),
-          )
+          this.setState({ isClear: e.target.value === '' })
+          let val = e.target.value
+          if (parseInt(val, 10) > 30) {
+            val = '30'
+          } else if (parseInt(val, 10) < 1) {
+            val = '1'
+          }
+          if (parseInt(e.target.value) > 0) {
+            this.props.handleSetDuration(
+              Duration.fromObject({
+                days: clamp(1, 30, parseInt(e.target.value, 10)),
+              }).toISO(),
+            )
+          }
```

Comment:
Invert and flatten this so that the `if` statement returns and the `handleSetDuration` runs outside

```js
if (failure condition) {
  return
}

// normal/success flow
this.props.handleSetDuration( ... )
```

Reply:
addressed and implemented.